internal/static/js/app.js

```diff
@@ -102,7 +111,31 @@ const attachPlots = () => {
     }
 }
 
-const updatePlots = () => {
+function throttle(func, delay) {
+    let initial = true;
+    let last = null;
+    let timer = null;
+    return function () {
+        const context = this;
+        const args = arguments;
+        if (initial) {
+            func.apply(context, args);
+            initial = false;
+            last = Date.now();
+        } else {
+            clearTimeout(timer);
+            timer = setTimeout(function () {
+                const now = Date.now();
+                if (now - last >= delay) {
+                    func.apply(context, args);
+                    last = now;
+                }
+            }, delay - (Date.now() - last));
+        }
+    }
+}
+
+const updatePlots = throttle(() => {
```

Comment:
why is this necessary?

Reply:
It can avoid repeated redrawing of DOM caused by browser energy saving mode. This will not affect normal users.
Described in #120